ember-app/app/components/issue/activities/issue-status.js

```diff
@@ -0,0 +1,79 @@
+import Ember from 'ember';
+
+var IssueStatusComponent = Ember.Component.extend({
+  isVisible: function(){
+    return this.get("status.total_count") > 0;
+  }.property("status.state"),
+
+  status: {
+    statuses: [],
+  },
+  fetchStatus: function(){
+    var repo = this.get("issue.repo.data.repo.full_name");
+    var number = this.get("issue.number");
+    var url = `/api/${repo}/issues/${number}/status`
+    var _self = this;
+    Ember.$.get(url).then(function(response){
+      _self.set("status", response);
+    });
+  }.on("init"),
+
+  statusContexts: function(){
+    return this.get("status.statuses").map(function(status){
+      return status.context;
+    }).uniq();
+  }.property("status.statuses.[]"),
+  commitStates: function(){
+    return this.get("status.statuses").map(function(status){
+      return status.state;
+    }).uniq();
+  }.property("status.statuses.[]"),
+  stateCounts: function(){
+    var counts = {};
+    this.get("status.statuses").forEach(function(status){
+      if(counts[status.state]){
+        counts[status.state] = counts[status.state] + 1;
+      } else {
+        counts[status.state] = 1;
+      }
+    });
+    return counts;
+  }.property("status.statuses.[]", "status.statuses.@each.state"),
+  aggregateState: function(){
```

Comment:
This may deviate from GitHub's status policy somewhat as the `status.state` payload is by default GitHubs determined aggregateState. The difference being if any errors occur on GitHub the state is automatically determined as a failure, even if there are checks pending.